### PR TITLE
Add mypy as static type checker

### PR DESCRIPTION
**Pull request by pmeier**

### CODE REVIEW

File: ragna/_api/orm.py
Comment (pavithraes) [inline ragna/_api/orm.py:1]: I'm seeing `error: Unused "type: ignore" comment  [unused-ignore]` on for all the type ignores in this file

File: ragna/_api/schemas.py
Comment (pavithraes) [inline ragna/_api/schemas.py:1]: "Column[UUID]"; expected "UUID" I see several similar+related errors of the form: `incompatible type "Column[UUID]"; expected "UUID"  [arg-type]` (also, for `bool` & `str`), on the `Document`, `Chat`, and `ChatMetadata` classes in this file

File: ragna/_api/database.py [hunk 1]
```diff
@@ -211,5 +210,5 @@ def update_chat(session: Session, user: str, chat: schemas.Chat) -> None:
 
 def delete_chat(session: Session, user: str, id: uuid.UUID) -> None:
     orm_chat = _get_orm_chat(session, user=user, id=id)
-    session.delete(orm_chat)
+    session.delete(orm_chat)  # type: ignore[no-untyped-call]
```
Comment (pavithraes) [inline ragna/_api/database.py:213]: This line also raises `error: Unused "type: ignore" comment  [unused-ignore]`

**Review by pavithraes — COMMENTED** [review]
Thanks, @pmeier! 

I ran mypy against this PR locally, and I've shared some errors that it raised. I have very brief experience with mypy, so please feel free to ignore or make updates as you see fit. :)

**Review by pavithraes — APPROVED** [review]

File: ragna/_api/schemas.py
Comment (pmeier) [reply ragna/_api/schemas.py:1]: This and the error below as well come from `sqlalchemy`. For that one needs to `pip install sqlalchemy-stubs`. I did so in CI, but forgot to make them explicit as part of the dev env. This is why CI was green here.

I made them explicit now. Thanks for the heads up!

**Pull request merged**